### DISCUSSION TITLE
Focal point too abstract; need ability to crop images per-instance

### DISCUSSION THREAD

**Discussion by alexgleason** [wagtail/wagtail#14112] [Ideas]
For example, a page model may display an image in a horizontal banner, then the same image is displayed on that page's index as a square thumbnail. There's no editorial ability to manage those separately.

I can envision two possible implementations:
1. Configure your page to have a thumbnail field and banner field separately. The client can choose to make them the same. Then, a crop can be applied _per-field_. That is, a "crop" button may appear next to "edit", "delete", "choose another" and apply that crop only to this instance in this field.
2. Allow multiple focal points to be set _per-image_. For instance, the client can crop each image for a "banner", "thumbnail", etc. then they are templated accordingly.

Any thoughts?

**Comment by timorieber** [comment]
I came across the same requirement several times, too. Usually we upload images with at least 1920px on the long edge, and depending on the output dimensions these images often offer a few details that can be cropped to stand alone.

Typical workarounds so far:
- play around with the focal point (editors)
- play around with the resizing options (developer)
- avoid multiple usage
- multiple uploads of the same image

All options are suboptimal, but could often be solved with multiple focal points. My first thoughts were similar to option 2 as suggested by @alexgleason. I would prefer this one over option 1, because the same focal point is simple to reuse with the exact same properties (ensures consistency across pages, minimizes renditions to be cached).

**Comment by hminnovation** [comment]
From an editor point of view I think option 1 may be easier to understand conceptually though, even if it does feel like it's leading to replication. I agree that option 2 feels more elegant, but I'm aware that there are editors that struggle with the concept of a single focal point let alone trying to introduce multiple. Would be exciting to see if we could make it work, but I think would be an interesting design/ UX challenge to pull off. @alexgleason how were you thinking you could communicate those multiple focal points for a single image?

**Comment by JoshBarr** [comment]
1. Sign up for [prismic.io](http://prismic.io)
2. Use the crop feature in there for a few minutes to see how it can be 100% awesome.
3. Implement here :) 

EDIT:

## [Cropping in prismic (youtube)](https://youtu.be/a8fl1Ml3llM)

**Comment by davecranwell** [comment]
Focal point is hard to explain to clients. Either their image isn't quite large enough for the use, or it's not sufficiently well composed for the focal point to have sufficient effect.

The duplication created by cropping is only a problem if you treat the crop and the original as separate images. They aren't - they're renditions of the same image, just as our existing concept of renditions doesn't create two records in the image library. You would be creating a lot more image files _on disk_, but we shouldn't be limiting users' control of their own assets for reasons like that.

I'd expect that the image chooser field would become two fields: 1. a pointer to the image record to use and 2. a pointer to a specific rendition of that image, created manually by the user (or defaulting to the whole image otherwise).

Of course, this does change how renditions requested by the  _developer_ work. If a developer needs a 16:9 image for a hero, this can no longer be defined at a template-level. The model needs to know, so the cropping system can guide the user to create an image that meets the developer's needs as much as the user's.

**Comment by tmsndrs** [comment]
I think there are a few issues here. 

*_UI *_
The focal point UI is potentially confusing in the way it shapes an editors expectations. It feel like a crop UI but actually all you're doing is defining a primary point of focus on the image. It would make more sense for the UI to use a cross hair rather than a marque to define the point IMO. 

*_Context *_
One of the issues with any tool like this is that it can encourage the editor to think about the image in a specific instance (which is always going to be the device they are on) but as the image is likely to look different on different devices a tool like this becomes pretty counter productive. Editors should be thinking about how their content could look across multiple devices and contexts and Wagtail should support that thinking not work against it.

*_Composition *_
I actually quite like Wagtail's current approach because it's asking you for metadata about the image (what is the main point of focus is true no matter what context that image might get used in) rather than how it should be used in a specific instance but the data is usually used with the assumption that the point of focus also related to how the image should be cropped which is an incorrect assumption from a composition perspective. Take for example a photo that uses the 'rule of thirds'. The focal point probably sits in the first or third section of the image. By treating the focal point like it's the centre of the image you're destroying the photos original composition, which should be an editorial decision rather than hard coded in the front end.

**Comment by tobiase** [comment]
I'd like to provide another example.

There's a drupal project called crop api: https://www.drupal.org/project/crop. You as a developer can choose, wether it should use a focal point or crop sizes, that are provided by a developer.
Watch https://www.youtube.com/watch?v=Ioo_bYnCOOg to get an idea of cropping using different crop sizes instead of focal points.

**Comment by luizboaretto** [comment]
I have this same requirement here too.

The solution for now... the editor sends the same image whenever it needs a different crop/point.

It would be nice if the editor could select the focal point per instance!  :+1:

**Comment by browze** [comment]
Hi, just submitted a pull request for this: https://github.com/wagtail/wagtail/pull/4714

**Comment by lb-** [comment]
Noting that this PR https://github.com/wagtail/wagtail/pull/4714 has been closed, but would be a good starting point for anyone exploring a new PR for this feature.

See also a potentially overlapping request for setting the focal point of an image when uploading https://github.com/wagtail/wagtail/issues/3695

**Comment by thibaudcolas** [comment]
I see this has stalled a lot and I’m not sure we’ve had much feedback that focal points were confusing over the last ±5 years. I’m pretty sure they could be less confusing, but I don’t see ourselves wanting to support both focal points and cropping. And focal points are just a much better approach.

Will convert to a discussion, see if others have more concrete ideas on how to improve the status quo.
